x/gamm/keeper/grpc_query.go

```diff
@@ -127,6 +127,27 @@ func (q Querier) NumPools(ctx context.Context, _ *types.QueryNumPoolsRequest) (*
 	}, nil
 }
 
+func (q Querier) PoolType(ctx context.Context, req *types.QueryPoolTypeRequest) (*types.QueryPoolTypeResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	pool, err := q.Keeper.GetPoolAndPoke(sdkCtx, req.PoolId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	switch pool := pool.(type) {
+	case *balancer.Pool:
+		return &types.QueryPoolTypeResponse{PoolType: "Balancer"}, nil
```

Comment:
We should have this be exposed as a method on the pool interface, right?

Or at least, lets make a go API for `keeper.PoolType(ctx, poolId) (string, error)`, to have it in a more discoverable point

Reply:
Changed to having go API, I would prefer having further discussion in adding pool type to `PoolI`, I'd like to go ahead and open an issue for this for further discussion for this if you agree!

Reply:
SGTM